pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+            return ({
+                justCurrent: current && !state.auto,
+                applyCheckbox: current,
+                showModal: false,
+                error: "",
+            });
+        });
+    }
+
+    handleChange() {
+        this.setState({ updating: true });
+
+        if (this.state.applyCheckbox) {
+            let install = Promise.resolve(true);
+            if (this.state.justCurrent) {
+                install = new Promise((resolve, reject) => {
+                    cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" })
+                            .then(() => {
+                                if (!this.state.patchUnavailable && !this.state.patchInstalled)
+                                    cockpit.spawn(["dnf", "-y", "install", this.state.patchName], { superuser: "require", err: "message" }).then(resolve)
+                                            .catch(reject);
+                                else
+                                    resolve();
+                            })
+                            .catch(reject);
+                });
+            } else {
+                install = cockpit.spawn(["dnf", "-y", "kpatch", "auto"], { superuser: "require", err: "message" });
+            }
+            install.then(() => {
+                this.kpatchService.enable().then(() => {
+                    this.kpatchService.start().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        } else {
+            cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" }).then(() => {
+                this.kpatchService.disable().then(() => {
+                    this.kpatchService.stop().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        }
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false || this.state.patchName === null)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
```

Comment:
At this point we already know that this.state.unavailable.length == 0. I think it makes sense to exit early (as in line 214).

Reply:
> At this point we already know that this.state.unavailable.length == 0

correct, I can just drop the second part of the check.

> I think it makes sense to exit early (as in line 214).

Exit early? We need to render the "Install" button.

Reply:
I meant that we keep the condition in 214 (the "early exit" code path).